### PR TITLE
E1729. Export scores in detail

### PR DESCRIPTION
**Pull request by kkushagra**
Changes to add new feature, which exports a detailed CSV.

The changes have been explained in wiki : http://wiki.expertiza.ncsu.edu/index.php/CSC/ECE_517_Spring_2017/oss_E1729

### CODE REVIEW

**Comment by Winbobob** [comment]
Hi team,

Please try to fix the issues (except `Cyclomatic complexity`, `Perceived complexity` and `Assignment Branch Condition size`) reported by code climate.

Thanks,
Zhewei

File: app/controllers/export_file_controller.rb [hunk 1]
```diff
@@ -22,21 +22,46 @@ def start
...
+    elsif delim_type == "tab"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = "\t"
-    elsif @delim_type == "other"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "other"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = other_char
     end
+    filename, delimiter
```
Comment (Winbobob) [inline app/controllers/export_file_controller.rb]: This line means this method returning 2 variables, right?

File: spec/models/assignment_spec.rb [hunk 2]
```diff
@@ -114,5 +114,78 @@
...
+    create(:question)
+    create(:review_response_map)
+    create(:response)
+  end
+
+  delimiter = ","
+
+  it "checks_if_csv has the correct data" do
+    create(:answer, comments: "Test comment")
+    options = {"team_id" => "true", "team_name" => "true",
```
Comment (Winbobob) [inline spec/models/assignment_spec.rb]: You can put options in `before(:each)` and set the only change in each test case, which should remove the dup code. And after that I will approve other code climate issues, and there will be a green check for your pull request.

File: app/controllers/export_file_controller.rb (hunk 1 above)
Comment (kkushagra) [inline app/controllers/export_file_controller.rb]: Yes.

File: spec/models/assignment_spec.rb [hunk 3]
```diff
@@ -114,5 +114,73 @@
...
+                "question_id" => "true", "comment_id" => "true",
+                "comments" => "true", "score" => "true"}
+  end
+
+  delimiter = ","
+
+  it "checks_if_csv has the correct data" do
+    create(:answer, comments: "Test comment")
+    expected_csv = File.read('spec/features/assignment_export_details/expected_details_csv.txt')
+    generated_csv = CSV.generate(col_sep: delimiter) do |csv|
```
Comment (Winbobob) [inline spec/models/assignment_spec.rb]: For identical code, just extract a new method and pass necessary params.

**Comment by Winbobob** [comment]
Hi team,

If you finish top 6 issues reported by code climate, I will approve the rest issues. And there will be a green check appear.

Thanks,
Zhewei

File: app/controllers/export_file_controller.rb [hunk 4]
```diff
@@ -22,21 +22,47 @@ def start
...
+    elsif delim_type == "tab"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = "\t"
-    elsif @delim_type == "other"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "other"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = other_char
     end
+    return filename, delimiter
```
Comment (kkushagra) [inline app/controllers/export_file_controller.rb]: The code climate complains about the redundant return statement but while returning two values the return statement is mandatory, else the controller throws an error.

Comment (kkushagra) [inline app/controllers/export_file_controller.rb]: All other code climate issues have been fixed. 
Kindly review it by going through our deployment, the manual testing part talks about how you can test the feature.

http://wiki.expertiza.ncsu.edu/index.php?title=CSC/ECE_517_Spring_2017/oss_E1729

Comment (Winbobob) [inline app/controllers/export_file_controller.rb]: I just approved remaining issues.

File: diff [hunk 5]
```diff
@@ -0,0 +1,205 @@
+diff --git a/app/models/assignment.rb b/app/models/assignment.rb
```
Comment (Winbobob) [inline diff]: You do not need to commit this file.

**Comment by kkushagra** [comment]
Deleted the dangling diff file.

**Pull request merged**